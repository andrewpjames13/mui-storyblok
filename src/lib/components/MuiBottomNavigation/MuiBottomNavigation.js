import React, { useState, createElement } from 'react';
import PropTypes from 'prop-types';
import { BottomNavigation, BottomNavigationAction } from '@material-ui/core';
import MuiIconButtonRedirect from '../MuiIconButtonRedirect/MuiIconButtonRedirect';
import MuiIconButton from '../MuiIconButton/MuiIconButton';
import StoryBlok from '../../utils/Storyblok';

const MuiBottomNavigation = ({
  rootClass,
  showLabels,
  navActions,
  width,
}) => {
  const components = {
    MuiIconButton,
    MuiIconButtonRedirect,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass, {
    width: parseInt(width, 10),
    position: 'absolute',
    bottom: 25,
  });
  const [value, setValue] = useState(0);

  return (
    <BottomNavigation
      value={value}
      onChange={(e, newValue) => {
        setValue(newValue);
      }}
      showLabels={showLabels}
      className={styles.root}
    >
      {navActions.map(item => (
        <BottomNavigationAction
          {...item}
          key={item.value}
          icon={createElement(
            components[item.icon[0].component],
            Object.assign(item.icon[0]),
          )}
        />
      ))}
    </BottomNavigation>
  );
};

export default MuiBottomNavigation;

MuiBottomNavigation.propTypes = {
  /**
   * storyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: boolean
   * If true, all BottomNavigationActions will show their labels.
   * By default, only the selected BottomNavigationAction will show its label.
   */
  showLabels: PropTypes.bool,
  /**
   * mui prop: number
   * Width of the bottom navigation bar.
   */
  width: PropTypes.string,
  /**
   * components:
   * MuiBottomNavigationAction
   */
  navActions: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiBottomNavigation.defaultProps = {
  rootClass: [],
  showLabels: false,
  width: '500',
};
