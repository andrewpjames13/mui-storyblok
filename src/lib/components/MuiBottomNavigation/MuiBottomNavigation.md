MuiBottomNavigation example:

```js
  const navActions = [{
    component: 'MuiBottomNavigationAction',
    icon: [{
      component: 'MuiIconButton',
      disableFocusRipple: true,
      disableRipple: true,
      icon: [{
        component: 'MuiIcon',
        iconName: 'home',
      }],
    }],
  },
  {
    component: 'MuiBottomNavigationAction',
    icon: [{
      component: 'MuiIconButton',
      disableFocusRipple: true,
      disableRipple: true,
      icon: [{
        component: 'MuiIcon',
        iconName: 'menu',
      }],
    }],
  },
  {
    component: 'MuiBottomNavigationAction',
    icon: [{
      component: 'MuiIconButton',
      disableFocusRipple: true,
      disableRipple: true,
      icon: [{
        component: 'MuiIcon',
        iconName: 'add',
      }],
    }],
  },
  {
    component: 'MuiBottomNavigationAction',
    icon: [{
      component: 'MuiIconButton',
      disableFocusRipple: true,
      disableRipple: true,
      icon: [{
        component: 'MuiIcon',
        iconName: 'accessibility',
      }],
    }],
  }];

  const rootClass = ['bottom: 0', 'position: relative'];

  <MuiBottomNavigation
    showLabels={true}
    width={600}
    navActions={navActions}
    rootClass={rootClass}
  />
```