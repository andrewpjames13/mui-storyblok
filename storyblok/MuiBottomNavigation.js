module.exports = {
  name: 'MuiBottomNavigation',
  display_name: null,
  updated_at: '2020-05-04T16:03:34.135Z',
  schema: {
    rootClass: {
      type: 'options',
      description: '* storyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component.',
    },
    showLabels: {
      type: 'boolean',
      description: '* mui prop: boolean\n* If true, all BottomNavigationActions will show their labels.\n* By default, only the selected BottomNavigationAction will show its label.',
    },
    navActions: {
      type: 'bloks',
      restrict_components: true,
      component_whitelist: [
        'MuiBottomNavigationAction',
      ],
    },
    width: {
      type: 'number',
      description: 'Width of the bottom navigation bar',
      default_value: '500',
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiBottomNavigation',
  component_group_uuid: null,
};
