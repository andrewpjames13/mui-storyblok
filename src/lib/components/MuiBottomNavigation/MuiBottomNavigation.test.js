import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiBottomNavigation from './MuiBottomNavigation';

function setup() {
  const props = {
    showLabels: false,
    width: '500',
    navActions: [
      {
        component: 'MuiBottomNavigationAction',
        label: 'home',
        value: 'home',
        icon: [
          {
            component: 'MuiIconButtonRedirect',
            redirectRoute: 'home',
            iconButton: [
              {
                component: 'MuiIconButton',
                icon: [
                  {
                    component: 'MuiIcon',
                    iconName: 'home',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  };

  const comp = mount(
    <MemoryRouter>
      <MuiBottomNavigation {...props} />
    </MemoryRouter>,
  );
  return { comp, props };
}

describe('<MuiBottomNavigation />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiBottomNavigation {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiBottomNavigation', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('should set a new value onChange', () => {
    const { comp } = setup();
    const { value } = comp.find('WithStyles(ForwardRef(BottomNavigation))').first().props();
    expect(value).toEqual(0);
    const btn = comp.find('[data-testid="muiIconButton"]');
    btn.first().simulate('click');
    const newValue = comp.find('WithStyles(ForwardRef(BottomNavigation))').first().props().value;
    expect(newValue).toEqual('home');
  });
});
