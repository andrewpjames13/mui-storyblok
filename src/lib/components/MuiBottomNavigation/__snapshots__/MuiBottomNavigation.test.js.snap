// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiBottomNavigation /> snapshot 1`] = `
<div
  className="MuiBottomNavigation-root makeStyles-root-41"
>
  <MuiBottomNavigationAction
    aria-disabled={false}
    className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-root-42"
        data-testid="muiIconButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root makeStyles-root-43 MuiIcon-colorSecondary"
          >
            home
          </span>
        </span>
      </button>
      <span
        className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly"
      >
        home
      </span>
    </span>
  </MuiBottomNavigationAction>
</div>
`;
